src/officehours_api/views.py

```diff
@@ -53,17 +53,34 @@ class UserList(DecoupledContextMixin, generics.ListAPIView):
     serializer_class = ShallowUserSerializer
 
 
-class UserDetail(DecoupledContextMixin, generics.RetrieveAPIView):
+class UserDetail(DecoupledContextMixin, LoggingMixin, generics.RetrieveUpdateAPIView):
     queryset = User.objects.all()
     permission_classes = (IsAuthenticated,)
 
-    def get(self, request, *args, **kwargs):
+    def get_serializer(self, instance, data=None, many=None, partial=None):
+        ctx = self.get_serializer_context()
+        kwargs = {}
+        if data:
+            kwargs['data'] = data
+        return (
+            MyUserSerializer(instance, context=ctx, **kwargs)
+            if instance == ctx['user']
+            else ShallowUserSerializer(instance, context=ctx, **kwargs)
+        )
```

Comment:
I saw this traceback when I was testing this with the Swagger UI. Looks like they offer a solution, but don't know exactly what's going on myself.
```
view's UserDetail raised exception during schema generation; use `getattr(self, 'swagger_fake_view', False)` to detect and short-circuit this
Traceback (most recent call last):
File "/usr/local/lib/python3.8/site-packages/drf_yasg/inspectors/base.py", line 50, in call_view_method
     return view_method()
TypeError: get_serializer() missing 1 required positional argument: 'instance'
```

Reply:
Good catch, I did fix this by setting `instance=None` in the args (made the arg optional).

Reply:
Nice